### PR TITLE
Tech/test not logged and not run

### PR DESCRIPTION
**Pull request by RomanenkoVladimir**
_**Please read the [CONTRIBUTING.md](https://github.com/MaibornWolff/codecharta/blob/main/CONTRIBUTING.md) before opening a PR.**_

Issue: #787 (connected to?)

## Description
Some tests in the backend are skipped due to us using a Junit5 test engine and attempting to run Junit4 tests on it.
This PR adresses the issue, when merged it will:

- Change Testing to Junit 5
  - enable legacy support, with Vintage Engine for SCMLogParser
- Add Logging for test output
- Update Mokk version to the current 1.10 to fix compatibility issues between Mokk and Corroutines
- remove deprecated gradle key words from scripts (still lack compatibility to gradle 7 due to used libraries)

### CODE REVIEW

File: analysis/import/UnderstandImporter/build.gradle [hunk 1]
```diff
@@ -14,6 +14,7 @@ dependencies {
 
     runtime group: 'org.slf4j', name: 'slf4j-simple', version: slf4j_version
 
+    testImplementation('org.junit.jupiter:junit-jupiter:5.6.0')
```
Comment (NearW) [inline analysis/import/UnderstandImporter/build.gradle:17]: Why do we need junit5 if we run the tests using spek2?
Isn't it enough to just include spek2?

File: analysis/filter/EdgeFilter/build.gradle [hunk 2]
```diff
@@ -27,7 +27,10 @@ dependencies {
 }
 
 test {
-    useJUnitPlatform {
+    useJUnitPlatform{
```
Comment (NearW) [inline analysis/filter/EdgeFilter/build.gradle:30]: Do we have formatting for `build.gradle` files?

**Review by NearW — COMMENTED** [review]
LGTM, just had a few questions.

File: analysis/filter/StructureModifier/build.gradle [hunk 3]
```diff
@@ -7,10 +7,13 @@ dependencies {
     testImplementation group: 'org.jetbrains.kotlin', name: 'kotlin-test', version: kotlin_version
     testImplementation project(path: ':model', configuration: 'testOutput')
 
-    testImplementation group: 'junit', name: 'junit', version: junit4_version
+    testImplementation('org.junit.jupiter:junit-jupiter:5.6.0')
```
Comment (BridgeAR) [inline analysis/filter/StructureModifier/build.gradle:10]: Please keep the former notation (it is also possible to use a variable for the version) and move the version out in a variable (if there's none already that could be used instead)

File: analysis/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
+#Wed Sep 23 13:22:07 CEST 2020
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-all.zip
```
Comment (BridgeAR) [inline analysis/gradle/wrapper/gradle-wrapper.properties:2]: I suggest to use a variable for the version here to keep the version up to date.

File: analysis/import/CSVImporter/build.gradle [hunk 5]
```diff
@@ -14,6 +14,7 @@ dependencies {
 
     runtime group: 'org.slf4j', name: 'slf4j-simple', version: slf4j_version
 
+    testImplementation('org.junit.jupiter:junit-jupiter:5.6.0')
```
Comment (BridgeAR) [inline analysis/import/CSVImporter/build.gradle:17]: The same as above

File: analysis/import/CrococosmoImporter/build.gradle [hunk 6]
```diff
@@ -6,9 +6,13 @@ dependencies {
     implementation group: 'io.fastjson', name: 'boon', version: boon_version
     implementation group: 'info.picocli', name: 'picocli', version: picocli_version
 
+    testImplementation('org.junit.jupiter:junit-jupiter:5.6.0')
```
Comment (BridgeAR) [inline analysis/import/CrococosmoImporter/build.gradle:9]: The same as above and this applies to multiple other files as well.

**Comment by RomanenkoVladimir** [comment]
Ok, so I found the/a? issue:
Currently in SCMLogParser on main (as well as SCMLogParserV2 in development), we run junit4 tests on a junit5 engine, that seems to be the cause for some tests being skipped. 
I switched to junit5 where it was easy to do and enabled legacy support elsewhere (SCMLogParser).

The good: The change would run all tests previously skipped due to not being runnable on the Junit5 test engine, sadly I am not sure how many those are but it affects SCMLogParser especially (previously most tests here seem to have been skipped, from what I could tell with manual experiments and logging)

The bad: Some tests fail, 3 tests in SCMLogParser, however they also fail on the main branch if you call them manually (without gradle), meaning they never actually worked in the first place, but travis will be red with this "fix"

Additional thoughts:
- While legacy support is enabled and supported by JUnit we have already a huge bloat of different testing libraries and a testing refactor PR would be very usefull in my opinion, this could also help us remove JUnit4 by changing the few tests accordingly

- Our gradle script is really broken in regards to testing, with too many libraries doing more or less the same to the point where even the IDEA fails with autoimports from time to time, I am not sure if  more tests are skipped due to conflicts or outdated engines or similar problems

**Comment by NearW** [comment]
> The bad: Some tests fail, 3 tests in SCMLogParser, however they also fail on the main branch if you call them manually (without gradle), meaning they never actually worked in the first place, but travis will be red with this "fix"

Well, at least we found out now. Let's fix them though.

> While legacy support is enabled and supported by JUnit we have already a huge bloat of different testing libraries and a testing refactor PR would be very usefull in my opinion, this could also help us remove JUnit4 by changing the few tests accordingly.

I absolutely agree. I'd use JUnit5.

**Comment by RomanenkoVladimir** [comment]
@NearW the SourceCodeParser was also using junit4 Tests before, so tests would have been skipped.
It is possible that this caused the behaviour you mentioned.

**Comment by RomanenkoVladimir** [comment]
I would propose merging this at its stage and remove Junit 4 entirely during a follow-up PR.
This would enable to test the backend properly again, also the new SCMLogParserV2 has a few Junit4 tests as well, therefore I would propose waiting with the change until the feature is merged.

**Review by NearW — APPROVED** [review]

**Pull request merged**